task/validate-api/build.rs

```diff
@@ -53,7 +96,35 @@ fn write_pub_device_descriptions(
         "pub static DEVICES: [DeviceDescription; DEVICES_CONST.len()] = DEVICES_CONST;"
     )?;
 
+    writeln!(
+        file,
+        "pub const DEVICE_INDICES_BY_ID_CONST: [([u8; MAX_ID_LENGTH], usize); {}] = [",
+        id2idx.len()
+    )?;
+    for (id, idx) in id2idx {
+        writeln!(file, "    ({id:?}, {idx}),")?;
+    }
+    writeln!(file, "];")?;
+    writeln!(
+        file,
+        "pub static DEVICE_INDICES_BY_ID: [([u8; MAX_ID_LENGTH], usize); \
+         DEVICE_INDICES_BY_ID_CONST.len()] = DEVICE_INDICES_BY_ID_CONST;"
```

Comment:
Do we need both `const` and `static` versions here?

Also, naming them something like `DEVICE_INDICES_BY_SORTED_ID` would make the `binary_search` call more obvious.

Reply:
ah, i thought something was using the const but on further inspection you're right, i removed it. good call about the naming!